base/nullable.jl

```diff
@@ -175,18 +175,39 @@ vectorization.
 """
 null_safe_op(f::Any, ::Type, ::Type...) = false
 
-typealias NullSafeSignedInts Union{Int128, Int16, Int32, Int64, Int8}
-typealias NullSafeUnsignedInts Union{Bool, UInt128, UInt16, UInt32, UInt64, UInt8}
+typealias NullSafeSignedInts Union{Type{Int128}, Type{Int16}, Type{Int32},
+                                   Type{Int64}, Type{Int8}}
+typealias NullSafeUnsignedInts Union{Type{Bool}, Type{UInt128}, Type{UInt16},
+                                     Type{UInt32}, Type{UInt64}, Type{UInt8}}
 typealias NullSafeInts Union{NullSafeSignedInts, NullSafeUnsignedInts}
-typealias NullSafeFloats Union{Float16, Float32, Float64}
+typealias NullSafeFloats Union{Type{Float16}, Type{Float32}, Type{Float64}}
 typealias NullSafeTypes Union{NullSafeInts, NullSafeFloats}
+typealias EqualOrLess Union{typeof(isequal), typeof(isless)}
+
+null_safe_op{T}(::typeof(identity), ::Type{T}) = isbits(T)
+
+@pure null_safe_eltype_op(op, a) = null_safe_op(op, eltype(a))
+@pure null_safe_eltype_op(op, a, b) = null_safe_op(op, eltype(a), eltype(b))
+@pure null_safe_eltype_op(op, a, b, c) =
+    null_safe_op(op, eltype(a), eltype(b), eltype(c))
+@generated function null_safe_eltype_op(op, xs...)
```

Comment:
>null_safe_op(op, map(eltype, xs)) is type-unstable, because xs is a tuple of DataType.

Wouldn't a strategy similar to [this](https://github.com/JuliaStats/NullableArrays.jl/blob/ba45e57691c42b7c1796a8b17a8f3f8dd80424bd/src/broadcast.jl#L79) work here too? (This is from first attempts at https://github.com/JuliaStats/NullableArrays.jl/pull/166, and it was correctly inferred).